### PR TITLE
export rsa public key as dem

### PR DESCRIPTION
**Pull request by liguopku**
@bhariharan export rsa public key as dem format

### CODE REVIEW

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SalesforceRestAPITests | testBlockUpdate|
SalesforceRestAPITests | testBlockUpdate|


# Tests results for SalesforceSDKCore

<p align="right" data-meta="generated_by_SalesforceSDKCore">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m [hunk 1]
```diff
@@ -424,4 +444,70 @@ +(nullable SecKeyRef)getRSAKeyRefWithTag:(NSString *)keyTagString keyLength:(NSU
     return keyRef;
 }
 
++ (NSData *)getRSAPublicKeyAsDER:(NSData *)keyData {
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m]: Did we originate this code?  If this is based on someone else's work, we need to attribute it, make sure we have a license for it, etc.

Comment (liguopku) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m]: I added the original blog link at comment header here but I may miss other places.
https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2392/files#diff-af9b3c572bb034509566bf659be0424fR95

Is there anywhere we also need to add additional comments?

Comment (khawkins) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSDKCryptoUtils.m]: No, I just missed it there.  Thanks for pointing it out, @liguopku !

**Pull request merged**